torch/csrc/generic/serialization.cpp

```diff
@@ -2,38 +2,46 @@
 #define TH_GENERIC_FILE "torch/csrc/generic/serialization.cpp"
 #else
 
-#ifdef THC_GENERIC_FILE
+#ifdef USE_CUDA
 #include <c10/cuda/CUDAGuard.h>
-#else
-#include <c10/core/CPUAllocator.h>
 #endif
 
+#include <c10/core/CPUAllocator.h>
+
 // save_save is necessary since the old eager format saved storages as
 // [size + data], but the v1.5 eager format removes this since size is saved in
 // the filesize.
 template <class io>
 void THPStorage_(writeFileRaw)(c10::StorageImpl *self, io fd, bool save_size, uint64_t element_size)
 {
-#ifdef THC_GENERIC_FILE
-  c10::cuda::CUDAGuard guard(self->device());
+#ifdef USE_CUDA
+  c10::cuda::OptionalCUDAGuard guard;
```

Comment:
I'll change this and the other `OptionalCUDAGuard` in this file to `OptionalDeviceGuard`